src/KubernetesClient/Informers/Cache/SimpleCache.cs

```diff
@@ -0,0 +1,196 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace k8s.Informers.Cache
+{
+    public class SimpleCache<TKey, TResource> : ICache<TKey, TResource>, ICacheSnapshot<TKey, TResource>
```

Comment:
I do not believe the cache should implement `ICacheSnapshot` as it does not satisfy the contact of a "snapshot of cache at a point in time".

Consider:
```csharp
ICache<int, int> cache = new SimpleCache<int, int>();
ICacheSnapshot<int, int> snapshot = cache;
cache.Add(42, 0);
snapshot.ContainsKey(42); // returns true when it should be false.
```

Reply:
This would be a weird way of working with this class, though from purist perspective you're right. I was trying to reuse the same object for both snapshotting and "live" (at least for SimpleCache implementation). I guess I can create a some kind of childview type object for SimpleCache snapshot. Wondering if others have opinion on this before I fix: @brendandburns @tg123 ?

Reply:
I would be hesitate to make assumptions about how a type is to be used, especially one that is public and implicitly convertible to an interface that it does not implement correctly.

Here's a Rider generated implementation:
```csharp
private sealed class SimpleCacheSnapshot : ICacheSnapshot<TKey, TResource>
{
    private readonly Dictionary<TKey, TResource> _items;
    public SimpleCacheSnapshot(SimpleCache<TKey,TResource> cache, long version)
    {
        _items = new Dictionary<TKey, TResource>(cache._items);
        Version = version;
    }
    public IEnumerator<KeyValuePair<TKey, TResource>> GetEnumerator() => _items.GetEnumerator();
    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
    public int Count => _items.Count;
    public bool ContainsKey(TKey key) => _items.ContainsKey(key);
    public bool TryGetValue(TKey key, out TResource value) => _items.TryGetValue(key, out value);
    public TResource this[TKey key] => _items[key];
    public IEnumerable<TKey> Keys => _items.Keys;
    public IEnumerable<TResource> Values => _items.Values;
    public long Version { get; }
}
```

Reply:
Ok, changed as recommended